src/expr/src/linear.rs

```diff
@@ -1250,6 +1255,23 @@ pub mod plan {
     }
 
     impl MfpPlan {
+        pub fn permute(&mut self, permutation: &[usize]) {
+            self.mfp.mfp.permute(
```

Comment:
To avoid code duplication it seems better to call `as_map_and_new_arity` here.

Reply:
yes, I just have to change the API slightly, but that is doable.

Reply:
Ah, the problem is that `expr` can't see `Permutation`. I figured it earlier and then promptly forgot about it! Well, if you figure out a way to do this, that'd be great, but if not I can live with it.

Reply:
I think it's best to move that computation into a utility function in `expr` that lives near the MFP stuff. New commit incoming now.